### PR TITLE
gh-3 | Adding Troubleshoot Command to deps

### PR DESCRIPTION
**Pull request by pavan-kalyan**

### CODE REVIEW

**Comment by pavan-kalyan** [comment]
@mjpitz Hi, I'm a little unsure on how to implement this. mainly,
- Whether this implementation of httpTroubleshootClient and interaction with gateway makes sense,
- How exactly does the information needs to be displayed especially for Health information, which is just a json string output currently.

**Comment by mjpitz** [comment]
So lets start with the command.

1. Let's call it `debug` instead of `troubleshoot`. It's shorter and easier to remember.
2. Let's put the command in a file under `/deps/internal/cmds/debug/debug.go`. All the logic should be able to be encapsulated in this file.
3. I wouldn't try to backport anything into the internal client interface. I'm betting that's going to change. This is likely going to be a one-off call so I would put the client in the same file as the command.
4. I'm pretty open to formatting. I think the general flow of information should be:
  * client info (os, arch, environment variables)
  * version info (client and server
  * server info (server, health)

**Comment by dilukshashamal** [comment]
lets start your code

**Comment by dilukshashamal** [comment]
achieve your code

File: deps/internal/client/client.go [hunk 1]
```diff
@@ -26,8 +28,21 @@ const (
 var (
 	protocol = or(os.Getenv(VariableProtocol), DefaultProtocol)
 	baseURL  = or(os.Getenv(VariableBaseURL), DefaultBaseURL)
+	systemOS = runtime.GOOS
+	arch     = runtime.GOARCH
```
Comment (mjpitz) [inline deps/internal/client/client.go:32]: I'd just reference these inline. No need to create intermediary variables for them.

File: deps/internal/client/client.go [hunk 2]
```diff
@@ -26,8 +28,21 @@ const (
...
 
+type SystemInfo struct {
+	Protocol string
+	BaseURL  string
+	Os       string
+	Arch     string
+}
+
+func (s SystemInfo) String() string {
+	return fmt.Sprintf("{Protocol: %v, BaseURL: %v, OS: %v, Arch: %v}", s.Protocol, s.BaseURL, s.Os, s.Arch)
```
Comment (mjpitz) [inline deps/internal/client/client.go:43]: preference for lowercase when outputting these to match the version semantic:

`{protocol: "", baseURL: "", os: "", arch: ""}`

File: deps/internal/cmds/debug/debug.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+func Command(version mux.Version) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "debug",
+		Short: "Output information helpful for debugging",
+		RunE: func(_ *cobra.Command, args []string) error {
+			// Printing Client environment variables
+			systemInfo := client.GetSystemInfo()
+			fmt.Println(systemInfo.String())
```
Comment (mjpitz) [inline deps/internal/cmds/debug/debug.go:22]: fmt.Println(client.GetSystemInfo())

File: deps/internal/cmds/debug/debug.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+				fmt.Println(fmt.Sprintf("Server Version: %s", serverVersion))
+			}
+			// Printing Server Health information
+			healthString, err := debugClient.GetHealth()
+			if err != nil {
+				fmt.Println(fmt.Sprintf("Error While retrieving server health"))
+			} else {
+				fmt.Println(fmt.Sprintf("Server Health: %s", healthString))
+			}
+			return nil
```
Comment (mjpitz) [inline deps/internal/cmds/debug/debug.go:44]: So I think this would read better if it followed the flow of:
* fetch information
* Output information

**Review by mjpitz — CHANGES_REQUESTED** [review]

File: deps/internal/cmds/debug/debug.go (hunk 4 above)
Comment (pavan-kalyan) [reply deps/internal/cmds/debug/debug.go:44]: Changed it to the mentioned flow

File: deps/internal/client/client.go (hunk 1 above)
Comment (pavan-kalyan) [reply deps/internal/client/client.go]: Changed

File: deps/internal/client/client.go (hunk 2 above)
Comment (pavan-kalyan) [reply deps/internal/client/client.go]: changed

File: deps/internal/cmds/debug/debug.go (hunk 3 above)
Comment (pavan-kalyan) [reply deps/internal/cmds/debug/debug.go]: changed

File: deps/internal/cmds/debug/debug.go (hunk 4 above)
Comment (pavan-kalyan) [reply deps/internal/cmds/debug/debug.go:44]: Changed it to the above mentioned flow

**Review by mjpitz — APPROVED** [review]
LGTM!

**Pull request merged**